src/lib/components/toast/ToastProgressBar.tsx

```diff
@@ -0,0 +1,45 @@
+import { darken } from 'polished';
+import { useEffect, useState } from 'react';
+import { ProgressBar } from '../progressbar/ProgressBar.component';
+
+export function ToastProgressBar({
```

Comment:
nit: I wonder wether we shouldn't enlarge the name of this component. It's more like a 'fake' progress bar based on a duration than a Toast specific progress bar
```suggestion
export function DurationBasedProgressBar({
```